pkgs/top-level/haskell-packages.nix

```diff
@@ -85,7 +88,10 @@ in {
       llvmPackages = pkgs.llvmPackages_9;
     };
     ghc8102 = callPackage ../development/compilers/ghc/8.10.2.nix {
-      bootPkgs = packages.ghc865Binary;
+      bootPkgs = if stdenv.isAarch64 then
+          packages.ghc8102Binary
+        else
+          packages.ghc865Binary;
```

Comment:
You might want to add a comment here as well explaining why this is necessary (so that someone doesn't accidentally try to change it in the future).